### ISSUE TITLE
tried instal with Kernel 6.8.6 on ubuntu, didnt work.

### ISSUE THREAD

**Issue by josepbw82** [morrownr/rtl8852bu#27]
DKMS make.log for rtl8852bu-1.19.3 for kernel 6.8.6-060806-generic (x86_64)
Sat Apr 13 07:08:37 PM EDT 2024
make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/6.8.6-060806-generic/build M=/var/lib/dkms/rtl8852bu/1.19.3/build  modules
make[1]: Entering directory '/usr/src/linux-headers-6.8.6-060806'

  ERROR: Kernel configuration is invalid.
         include/generated/autoconf.h or include/config/auto.conf are missing.
         Run 'make oldconfig && make prepare' on kernel src to fix it.

make[2]: *** [/usr/src/linux-headers-6.8.6-060806/Makefile:798: include/config/auto.conf] Error 1
make[1]: *** [Makefile:240: __sub-make] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-6.8.6-060806'
make: *** [Makefile:746: modules] Error 2

**Comment by josepbw82** [comment]
"make oldconfig" did not give error.
This is what happened when i did "make prepare" 
make[2]: *** No rule to make target 'arch/x86/entry/syscalls/syscall_32.tbl', needed by 'arch/x86/include/generated/uapi/asm/unistd_32.h'.  Stop.
make[1]: *** [arch/x86/Makefile:249: archheaders] Error 2
make: *** [Makefile:240: __sub-make] Error 2

**Comment by morrownr** [comment]
This the beta of 24.04? Right?

Recommend you report this to Ubuntu. I won't be using 24.04 until sometime after it is released. I have tested kernel 6.8 and it works well but I tested on 23.10. It looks like 24.04 is broken.

**Comment by josepbw82** [comment]
Well, it's the ubuntu mainline kernel 6.8.6 on Ubuntu 23.10

I installed on the mainline kernel 6.5, either way, Network manager and iwd
did not show the device.
When I search in /sys, the device shows. Lsusb shows the device and lshw
shows the Asus USB-NANO AX56. Scratching my head as this makes no sense.
Also doesn't recognize two other wifi dongles anymore. Might just be my
computer could be messed up and need to do a full wipe and full reinstall.
Smh.

**Comment by morrownr** [comment]
> Might just be my computer could be messed up and need to do a full wipe and full reinstall.

This could be the case but the problem looks like a header files problem so it is likely an Ubuntu problem.

If you are not a fan of compiling drivers that end up giving you a hard time for one reason of another:

https://github.com/morrownr/USB-WiFi

That is the Main Menu for this site. You may find menu items 1 and 2 to be helpful. We have good alternatives to Realtek's poor usb wifi adapter support.

**Comment by josepbw82** [comment]
Ok, so, now the driver works...just one problem.
lsusb
Appears at bus 005 device 008
Then disappears and comes back as bus005 device 009
It continues to disappear and reappear at the same bus_id and higher
device_id as if it were an incremental counter.
The root hub is bus 005 device 001
The other USB dongle on that hub is bus 005 device 002
Shouldn't the Asus ax wlan adaptor start at and remain as bus 005 device 003
Scratching my head, because I've never seen this happen before.

**Comment by morrownr** [comment]
> Shouldn't the Asus ax wlan adaptor start at and remain as bus 005 device 003. Scratching my head, because I've never seen this happen before.

This is likely a problem due to the adapter being multi-state. Multi-state means that the adapter has an onboard Windows driver and the adapter will initially come up as a storage device and has to have its mode switched to wifi. I try to make it clear that Linux users should avoid multi-state adapters in my document on the Main Menu in item 1 but I know that not everyone knows to read it before buying. I posted the link to the Main Menu previously in this thread.

On the Main Menu here I have a menu item that explains Modeswitching. here is a direct link:

https://github.com/morrownr/USB-WiFi/blob/main/home/How_to_Modeswitch.md

My recommendation is to go to the bottom on the document and deactivate usb_modeswitch so that we can see what is going on. The problem is likely that the information for your adapter is not exactly right and will need to be fixed.

Keep in mind that there are many really good usb wifi adapters that work well on Linux without this hassle so you need to think about how much crap you are willing to put up with before you make a change. Here is a direct link to the Plug and Play List. 

https://github.com/morrownr/USB-WiFi/blob/main/home/USB_WiFi_Adapters_that_are_supported_with_Linux_in-kernel_drivers.md

Note for Linux users that may read this someday: Realtek based WiFi 6 USB WiFi adapters are something you really want to avoid whether for home, office or embedded use. All of the Realtek based wifi adapters that I am aware of are multi-state and the Realtek drivers are really bad. The driver in this repo does client (managed) mode reasonably well after I spent a few months working on it. The other modes... well, good luck. There is no good reason to get a Realtek based WiFi 6 adapter if you can get a Mediatek based WiFi 6 adapter. Mediatek does a good job with their drivers and they do their drivers in accordance with Linux Wireless Standards.

**Comment by josepbw82** [comment]
I like how you worded all that. LoL.
Ok, I went through the usb_modeswitch and that didn't work for me.
I am currently looking at the windows driver and in file, which shows
8852au has to have a couple parameters set, and then the 8852bu section is
showing all available parameters and settings listed that can be used.  I
think I'll use the upcoming weekend to get all the options,settings, and
parameters listed into a readme file.  I think I'll have fun wasting my
time this weekend. LoL.

**Comment by josepbw82** [comment]
Wow, I started with Linux in 1998. A copy of Red hat Linux was given to me
by a classmate.  It has changed a lot since then.

**Comment by morrownr** [comment]
>  It has changed a lot since then.

Yes it has. I have noticed some changes over the last few years indicating stabilization in that we now see much less need to lock in a kernel for a long period to have a stable system. I'm running Debian 12 with kernel 6.6 provided by Debian. You even see this in Android. It is unfortunate that we USB WiFi adapters users have only two companies providing usb chips and one of them is stuck in the past. 

I very often have users ask me why I don't take the Realtek drivers here at this site to the kernel and I have to answer that it would be rejected as these out-of-kernel Realtek drivers do not meet modern Linux Wireless Standards. I really get tired of being asked and my answer is basically that we can control what we can control and Linux USB WiFi users should be buying adapters with Mediatek chipsets... as you can see in the Plug and Play List here on the Main Menu.

Anyway, let me know if I can help more. I have been unable to duplicate this problem so without more info, there is little I can do.

**Comment by josepbw82** [comment]
Well, some of the plug and play adapters won't work even though the
adaptors did a few months ago. Works on the other computer but not this
one.

**Comment by morrownr** [comment]
> Well, some of the plug and play adapters won't work even though the
adaptors did a few months ago...

What adapters are we talking about?
